From f481af736054c55bd4d8997a04a8cdfa6d1f5e97 Mon Sep 17 00:00:00 2001
From: "an.xi" <an.xi@amlogic.com>
Date: Tue, 24 Aug 2021 14:17:23 +0800
Subject: [PATCH] SystemServer: add aml hal traces and complete the file [1/2]

PD#SWPL-58413

Problem:
1.the buffer size is 192KB, it's too little which could
cause the stack is truncated most of time.
2.amlogic hal services should be added in the array of
Watchdog to help analyse stability issues.

Solution:
1.Enlarge the dropbox file buffer.
2.Add aml hal services in dropbox array.

Verify:
Ohm

Change-Id: I34272cd6f32908428c3f1af4f589dedfc0314be2
Signed-off-by: an.xi <an.xi@amlogic.com>
---
 services/core/java/com/android/server/Watchdog.java    | 10 ++++++++++
 .../com/android/server/am/ActivityManagerService.java  |  2 +-
 2 files changed, 11 insertions(+), 1 deletion(-)

diff --git a/services/core/java/com/android/server/Watchdog.java b/services/core/java/com/android/server/Watchdog.java
index 425a0452b6b8..1cbf42ddf03a 100644
--- a/services/core/java/com/android/server/Watchdog.java
+++ b/services/core/java/com/android/server/Watchdog.java
@@ -102,6 +102,16 @@ public class Watchdog extends Thread {
         "media.swcodec", // /apex/com.android.media.swcodec/bin/mediaswcodec
         "com.android.bluetooth",  // Bluetooth service
         "/apex/com.android.os.statsd/bin/statsd",  // Stats daemon
+        "/system/bin/hwservicemanager",
+        "/system/bin/servicemanager",
+        "/system/bin/imageserver",
+        "/vendor/bin/systemcontrol",
+        "/vendor/bin/tvserver",
+        "/vendor/bin/hdmicecd",
+        "/vendor/bin/hw/android.hardware.tv.cec@1.0-service",
+        "/vendor/bin/hw/android.hardware.tv.input@1.0-service",
+        "/vendor/bin/hw/android.hardware.graphics.composer@2.3-service.droidlogic",
+        "/vendor/bin/hw/android.hardware.graphics.allocator@3.0-service",
     };
 
     public static final List<String> HAL_INTERFACES_OF_INTEREST = Arrays.asList(
diff --git a/services/core/java/com/android/server/am/ActivityManagerService.java b/services/core/java/com/android/server/am/ActivityManagerService.java
index 28c9e4914e68..70c0bcfbb72b 100644
--- a/services/core/java/com/android/server/am/ActivityManagerService.java
+++ b/services/core/java/com/android/server/am/ActivityManagerService.java
@@ -494,7 +494,7 @@ public class ActivityManagerService extends IActivityManager.Stub
     static final String[] EMPTY_STRING_ARRAY = new String[0];
 
     // How many bytes to write into the dropbox log before truncating
-    static final int DROPBOX_DEFAULT_MAX_SIZE = 192 * 1024;
+    static final int DROPBOX_DEFAULT_MAX_SIZE =  1 * 1024 * 1024;
     // Assumes logcat entries average around 100 bytes; that's not perfect stack traces count
     // as one line, but close enough for now.
     static final int RESERVED_BYTES_PER_LOGCAT_LINE = 100;
-- 
2.26.1

